models/comments.models.js

```diff
@@ -15,4 +15,24 @@ exports.insertCommentByArticleId = (article_id, newComment) => {
         .then(({rows}) => {
             return rows[0]
         })
+}
+
+exports.fetchCommentByCommentId = (comment_id) => {
+    return db
+    .query("SELECT * FROM comments WHERE comment_id = $1", [comment_id])
+    .then(({rows}) => {
+        const comment = rows[0];
+        if(!rows.length){
+            return Promise.reject({
+                status: 404,
+                msg: `No comment found for comment id: ${comment_id}`
+            })
+        }
+        return comment;
+    })
+}
+
+exports.removeCommentById = (comment_id) => {
+    return db
+    .query("DELETE FROM comments WHERE comment_id = $1", [comment_id])
```

Comment:
Very good. You could have potentially saved yourself from doing a check-if-exists sql request by simply checking how many rows your delete request impacted. If it was 0, you could have thrown your error. res.rowCount is the property to check in the then block, I believe 🙂 

That being said, what you've done works well so it's up to you 🙂

Reply:
oh perfect that actually sounds really good because i was trying to figure out a way to check that something is actually deleted so i will definitely be refactoring this in after i've added my other endpoints! thanks so much for this 🥳